powertools-e2e-tests/src/test/java/software/amazon/lambda/powertools/IdempotencyE2ET.java

```diff
@@ -0,0 +1,64 @@
+package software.amazon.lambda.powertools;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import software.amazon.lambda.powertools.testutils.Infrastructure;
+import software.amazon.lambda.powertools.testutils.lambda.InvocationResult;
+
+import java.time.Year;
+import java.util.HashMap;
+import java.util.concurrent.TimeUnit;
+
+import static software.amazon.lambda.powertools.testutils.lambda.LambdaInvoker.invokeFunction;
+
+public class IdempotencyE2ET {
+    private static Infrastructure infrastructure;
+    private static String functionName;
+
+    @BeforeAll
+    @Timeout(value = 5, unit = TimeUnit.MINUTES)
+    public static void setup() {
+        infrastructure = Infrastructure.builder()
+                .testName(IdempotencyE2ET.class.getSimpleName())
+                .pathToFunction("idempotency")
+                .idempotencyTable("idempo")
+                .environmentVariables(new HashMap<>() {{
+                        put("IDEMPOTENCY_TABLE", "idempo");
+                    }}
+                )
+                .build();
+        functionName = infrastructure.deploy();
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        if (infrastructure != null)
+            infrastructure.destroy();
+    }
+
+    @Test
+    public void test_ttlNotExpired_sameResult_ttlExpired_differentResult() throws InterruptedException {
+        // GIVEN
+        String event = "{\"message\":\"TTL 10sec\"}";
+
+        // WHEN
+        // First invocation
+        InvocationResult result1 = invokeFunction(functionName, event);
+
+        // Second invocation (should get same result)
+        InvocationResult result2 = invokeFunction(functionName, event);
+
+        Thread.sleep(12000);
+
+        // Third invocation (should get different result)
+        InvocationResult result3 = invokeFunction(functionName, event);
+
+        // THEN
+        Assertions.assertThat(result1.getResult()).contains(Year.now().toString());
+        Assertions.assertThat(result2.getResult()).isEqualTo(result1.getResult());
```

Comment:
The second assertion here fails - the contents of the result are an exception thrown in the lambda invocation:

```java
'java.lang.Object com.fasterxml.jackson.databind.ObjectReader.readValue(java.lang.String, java.lang.Class)': java.lang.NoSuchMethodError
java.lang.NoSuchMethodError: 'java.lang.Object com.fasterxml.jackson.databind.ObjectReader.readValue(java.lang.String, java.lang.Class)'
	at software.amazon.lambda.powertools.idempotency.internal.IdempotencyHandler.handleForStatus(IdempotencyHandler.java:130)
	at software.amazon.lambda.powertools.idempotency.internal.IdempotencyHandler.processIdempotency(IdempotencyHandler.java:83)
	at software.amazon.lambda.powertools.idempotency.internal.IdempotencyHandler.handle(IdempotencyHandler.java:62)
	at software.amazon.lambda.powertools.idempotency.internal.IdempotentAspect.around(IdempotentAspect.java:65)
	at software.amazon.lambda.powertools.e2e.Function.handleRequest(Function.java:44)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
```

All the other assertions in this test and all the other tests in the suite pass

Reply:
The idempotency record is placed in the DDB table by the first call and looks reasonable:
```json
{
  "id": {
    "S": "IdempotencyE2ET-ec1222eea3fb-function.handleRequest#82cbedb09f8dc00076b49bf8ce1a3615"
  },
  "data": {
    "S": "\"2023-02-10T12:26:44.363455Z[UTC]\""
  },
  "expiration": {
    "N": "1676032014"
  },
  "status": {
    "S": "COMPLETED"
  }
}
```

Reply:
Potentially because of [this issue.](https://github.com/awslabs/aws-lambda-powertools-java/issues/1026) ?

Reply:
More details - fails locally with:
* SDK=corretto 18.02, language level=11
*

Reply:
The frame handleForStatus in IdempotencyHandler.java:130 changed as part of this PR -->

https://github.com/awslabs/aws-lambda-powertools-java/pull/970/files#diff-9bdee74cc329a3af6ffd6a812ebfe7a07f53e170673b499623f3a3ac937d93b2R130

Reply:
I change to the latest powertools (1.14) and don't have this anymore (actually I didn't have this at the time of pushing this PR.. I don't push non-working code). 
Also line 130 does not match with the error you have which is on jackson `readValue` (line 132). I also tried with corretto 18 and it's ok. 

And this piece of code:
``` 
if (returnType.equals(String.class))
                return record.getResponseData();
```
was actually added to avoid this error, if I remember correctly...